boards/nucleo_f429zi/src/main.rs

```diff
@@ -263,36 +267,53 @@ pub unsafe fn reset_handler() {
 
     // Clock to Port A is enabled in `set_pin_primary_functions()`
     let led_pins = static_init!(
-        [(&'static stm32f4xx::gpio::Pin, capsules::led::ActivationMode); 1],
+        [(&'static kernel::hil::gpio::Pin, capsules::led::ActivationMode); NUM_LEDS],
         [(
-            &stm32f4xx::gpio::PinId::PB07.get_pin().as_ref().unwrap(),
+            stm32f4xx::gpio::PinId::PB07.get_pin().as_ref().unwrap(),
             capsules::led::ActivationMode::ActiveHigh
         )]
     );
     let led = static_init!(
-        capsules::led::LED<'static, stm32f4xx::gpio::Pin<'static>>,
-        capsules::led::LED::new(led_pins)
+        capsules::led::LED<'static>,
+        capsules::led::LED::new(&led_pins[..])
     );
 
     // BUTTONs
     let button_pins = static_init!(
-        [(&'static stm32f4xx::gpio::Pin, capsules::button::GpioMode); 1],
-        [(
-            &stm32f4xx::gpio::PinId::PC13.get_pin().as_ref().unwrap(),
-            capsules::button::GpioMode::LowWhenPressed
-        )]
+        [&'static kernel::hil::gpio::InterruptPin; NUM_BUTTONS],
+        [stm32f4xx::gpio::PinId::PC13.get_pin().as_ref().unwrap()]
     );
+
+    let values = static_init!(
+        [kernel::hil::gpio::InterruptValueWrapper; NUM_BUTTONS],
+        [kernel::hil::gpio::InterruptValueWrapper::new()]
+    );
+    
+    for i in 0..NUM_BUTTONS {
+        let pin = button_pins[i];
+        let value = &values[i];
+        pin.set_client(value);
+        value.set_source(pin);
+    }
+
+    let config_value = static_init!(
+        [(&'static kernel::hil::gpio::InterruptValuePin, capsules::button::GpioMode); NUM_BUTTONS],
+        [(&values[0], capsules::button::GpioMode::LowWhenPressed)]
+    );
+
     let button = static_init!(
-        capsules::button::Button<'static, stm32f4xx::gpio::Pin>,
+        capsules::button::Button<'static>,
         capsules::button::Button::new(
-            button_pins,
+            &config_value[..],
             board_kernel.create_grant(&memory_allocation_capability)
         )
     );
-    for &(btn, _) in button_pins.iter() {
-        btn.set_client(button);
+
+    for i in 0..NUM_BUTTONS {
```

Comment:
prefer iterators over ranges

Reply:
Generally, yes, but in this case I was trying to keep a parallel structure to the earlier initialization code that iterated over both values and button_pins.